### ISSUE TITLE
ML-Benchmark prediction submission: Rossby-UNet-orog

### ISSUE THREAD

**Issue by mikhail-a-ivanov** [WCRP-CORDEX/ml-benchmark#36]
### Institution ID

SMHI

### Institution Name

Swedish Meteorological and Hydrological Institute

### Emulator identifier (`emulator_id`)

Rossby-UNet-orog

### Emulator Description

A UNet-based machine learning univariate downscaling model with five encoding and five decoding layers, as well as a bottleneck layer and a convolutional block attention module (CBAM) in the middle. The number of channels in the input layer is 16, a GroupNorm normalization (16 channels per group) is applied in every UNet layer, a dropout of 0.1 is used in every encoding layer and a higher dropout of 0.3 is used in the bottleneck. SiLU activation function is used across the whole network.

Predictors and predictands are (de)standardized using local (per gridpoint) monthly climatologies of the training period. The standardization is performed offline, while the destandardized outputs are produced directly during the inference time. When destandardizing the output of the precipitation model, all negative values are set to zero. The predictors are interpolated bicubically to the predictand (high-resolution) grid in the input layer of the UNet. They are further complemented with high-resolution coordinate grid scaled to [0, 1] range and time encodings. High resolution orography, scaled to [0, 1] range, can be added to the predictors as well. The last four years of the training set (1977-1980 for the ESD_pseudo_reality and 2096-2099 for the Emulator_hist_future) are reserved for validation. 

A combined MSE-FFT loss function is used to train both precipitation and maximum temperature models. MSE and FFT contributions are weighed equally. FFT loss contribution is computed by taking a MSE of the magnitude and phase components of the 2D discrete Fourier transforms of the model output and the ground truth. 

AdamW optimizer with a maximum learning rate of 1E-4, a weight decay of 1E-4 and a batch size of 64 is used to train the model for a total of 25 epochs. OneCycleLR scheduler is used to control the learning rate during the training. 

### Hardware and Training Details

Hardware: single NVIDIA A40 GPU (48 GB). Training time: <1 minute per epoch, ~10 minutes per model training (25 epochs). Inference time: ~1 second per year of predictions.

### Stochastic/Probabilistic Output

no

### Reference URL

To be added (the GitHub repo with the code will be kept private until the publication is submitted, but the access can be given upon request)

### Repository for Reproducibility

https://github.com/fuentesfranco/detex/tree/cordex-ml
